app/Http/Controllers/ScoresController.php

```diff
@@ -95,27 +92,23 @@ public function download($rulesetOrSoloId, $id = null)
                     ->firstOrCreate(['year_month' => $currentMonth], ['count' => 0])
                     ->incrementInstance('count');
 
-                if ($score instanceof ScoreBest) {
-                    $score->replayViewCount()
-                        ->firstOrCreate([], ['play_count' => 0])
-                        ->incrementInstance('play_count');
-                }
+                $score->legacyScore?->replayViewCount()
+                    ->firstOrCreate([], ['play_count' => 0])
+                    ->incrementInstance('play_count');
 
-                if ($soloScore !== null) {
-                    ScoreReplayStats
-                        ::createOrFirst(['score_id' => $soloScore->getKey()], ['user_id' => $soloScore->user_id])
-                        ->incrementInstance('watch_count');
-                }
+                ScoreReplayStats
+                    ::createOrFirst(['score_id' => $score->getKey()], ['user_id' => $score->user_id])
+                    ->incrementInstance('watch_count');
             }
         }
 
-        static $responseHeaders = [
-            'Content-Type' => 'application/x-osu-replay',
-        ];
-
-        return response()->streamDownload(function () use ($file) {
-            echo $file;
-        }, $this->makeReplayFilename($score), $responseHeaders);
+        return response()->streamDownload(
+            function () use ($file) {
+                echo $file;
+            },
+            "replay-{$score->getMode()}_{$score->beatmap_id}_{$score->getKey()}.osr",
```

Comment:
I don't think we should be using exactly the same naming format as legacy replay when the score ids will be different